src/plugins/data/public/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -76,16 +78,18 @@ export class IndexPatterns {
     return this.savedObjectsCache.map(obj => obj?.attributes?.title);
   };
 
-  getFields = async (fields: string[], refresh: boolean = false) => {
+  getFields = async (fields: IndexPatternCachedFieldType[], refresh: boolean = false) => {
```

Comment:
I prefer to rename this method to `getByIdOrTitle`  or something like that instead of introducing the`IndexPatternCachedFieldType` type

It allows us to get exception if we use it with different attributes.

Reply:
Agree to improve in later improvement prs. For this fix, would like to have as minimal changes as possible and scoped to 1 file if possible

Reply:
https://github.com/elastic/kibana/issues/56900